pkg/arvo/sys/hoon.hoon

```diff
@@ -11534,114 +11557,175 @@
     ++  apse
+      %+  knee  *help  |.  ~+
       ;~  pose
-        %+  cook  |=([a=term b=cord] %*(. *whit def (my [a b ~] ~)))
-        (exit fine)
-      ::
-        %+  cook  |=(a=cord %*(. *whit boy `[a ~]))
+        %+  cook  |=(a=cord %*(. *help summary.crib a))
+        ::TODO: if there is a $link then interpret it as the start
+        ::of a prefix comment instead of postfix
+        ::;~(less (exit ;~(plug (plus en-link) col ace)) (exit line))
         (exit line)
       ::
-        (easy *whit)
+        (easy *help)
       ==
     ::
-    ::
-    ++  beer
-      |=  $:  lab=(unit term)
-              boy=(unit (pair cord (list sect)))
-              def=(list (pair (pair term cord) (list sect)))
+    ++  leap                                            ::  whitespace less docs
+      %+  cold  ~
+      %-  plus
+      ;~  plug
+        (star ace)
+        ;~  pose
+          ;~(plug (just '\0a') (star ace))
+          ;~  plug  col  col
+            ;~(less ;~(pose larg smol) ;~(plug (star prn) (just '\0a')))
           ==
-      ^-  whit
-      =;  def  [lab boy (malt def) ~]
-      (turn def |=([[a=term b=cord] c=(list sect)] [a [b c]]))
-    ::
-    ::
-    ++  body
-      ;~  pose
-        ;~  plug                                        :: can duplicate ::
-          (into ;~(pfix (punt ;~(plug null col gar step)) line))
-          (easy ~)
         ==
+      ==
+    ::
+    ::  +smol: 2 aces then summary, 4 aces then paragraphs.
+    ++  smol
+      ;~  pfix
+        step
         ;~  plug
-          (into ;~(pfix step line))
-          (rant text)
+          ;~  plug
+            (plus en-link)
+            ;~(sfix ;~(pfix col ace (cook crip (star prn))) (just '\0a'))
+          ==
+          ;~(pfix null (rant ;~(pfix step step text)))
         ==
       ==
     ::
-    ++  text  (pick line code)                          ::  text line
-    ++  line  ;~(less ace (cook crip (star prn)))       ::  prose line
-    ++  code  ;~(pfix step step (cook crip (star prn))) ::  code line
-    ++  noel  ;~(plug (punt ;~(pfix step hax)) null)    ::  header padding
-    ++  head  ;~(pfix hax step cen sym)                 ::  header line
-    ++  null  (cold ~ (star ace))                       ::  blank line
-    ++  fine                                            ::  definition line
-      ;~  (glue ;~(plug col ace))
-        sym
-        (cook crip (star prn))
+    ::  +larg: 4 aces then summary, 2 aces then paragraphs.
+    ++  larg
+      ;~  pfix
+        step  step
+        ;~  plug
+          ;~  sfix
+            ;~  plug
+              ;~  pose
+                ;~(sfix (plus en-link) col ace)
+                ;~(less ace (easy *cuff))
+              ==
+              (cook crip (star prn))
+            ==
+            (just '\0a')
+          ==
+          ;~(pfix null (rant ;~(pfix step teyt)))
+        ==
       ==
     ::
+    ++  myth
+      |*  bod=rule
+      (cook |=([[a=cuff b=cord] c=(list sect)] [a b c]) bod)
```

Comment:
Yeah I feel like I'm doing something wrong elsewhere that makes this even necessary. I know I tried to solve this problem at least a couple times and failed, surely something like this would have been built already if it were actually needed. Your definition is much better in any case. I'll give it another look too try and get rid of this since it seems solvable and I'll probably never be this knowledgable about the Hoon parser again.